Add additional pkg-config support

hughmcmaster opened the pull request:
ICU is detected through pkg-config nowadays.

The second commit adds support for detecting libxslt via pkg-config as well.

Almost all of the complex detection code should be removed, as the functions looked for have been around for many years and pkg-config is very much preferred over the legacy foo-config scripts. I’m happy to make these changes if you like.

dajobe commented:
This patch fails in Travis CI when compiled with c++ which I used as a more advanced C compiler check.  Something related to templates

hughmcmaster commented:
I saw you'd fixed the templates issue.

I've rebased my patches on the master branch and fixed an error I found in my libxslt patch.

All Travis CI builds pass now.

dajobe commented on configure.ac:
```diff
@@ -571,6 +571,53 @@ if test "X$xslt_config" != "Xno" ; then
   fi
 fi
 
+libxslt_source=no
```
I think libxslt_source is never changed so the code below (and this line) are redundant

hughmcmaster replied:
`libxslt_source` was changed in the 'action if present' part of the PKG_CHECK_MODULES call. The code sequence serves not useful purpose, so I removed it.

dajobe commented on configure.ac:
```diff
@@ -571,6 +571,53 @@ if test "X$xslt_config" != "Xno" ; then
+  AC_MSG_CHECKING(for libxslt via pkg-config)
+  if test "$libxslt_source" != "no"; then
+    have_libxslt=1
+    AC_MSG_RESULT(yes - $LIBXSLT_VERSION)
```
The if for libxslt_source will always fail so it is not needed.  612-618 can be replaced with
```
have_libxslt_=0
AC_MSG_RESULT(no - not found)
```

hughmcmaster replied:
I still feel much of configure.ac is overly complicated, particularly with libxml and libxslt. The minimum libxslt version has been around for 18 years. So, we could easily cut the function and header checks. If you're interested, that would be a separate PR.

dajobe commented:
Looks great now thanks
